test/models/podcast_planner_test.rb

```diff
@@ -0,0 +1,423 @@
+      assert_equal planner.ready_to_select_weeks?, false
+      planner.monthly_weeks = [1]
+      assert_equal planner.ready_to_select_weeks?, true
+      planner.monthly_weeks = [1, 2]
+      assert_equal planner.ready_to_select_weeks?, true
+    end
+
+    it "ready_to_select_date_range?" do
+      assert_equal planner.ready_to_select_date_range?, false
+
+      planner.date_range_condition = "episodes"
+      assert_equal planner.ready_to_select_date_range?, false
+      planner.number_of_episodes = 10
+      assert_equal planner.ready_to_select_date_range?, true
+
+      planner.date_range_condition = "date"
+      assert_equal planner.ready_to_select_date_range?, false
+      planner.start_date = DateTime.new(2001, 2, 3)
+      planner.end_date = DateTime.new(2001, 2, 1)
+      assert_equal planner.ready_to_select_date_range?, false
+      planner.end_date = DateTime.new(2001, 2, 3)
+      assert_equal planner.ready_to_select_date_range?, false
+      planner.end_date = DateTime.new(2001, 2, 4)
+      assert_equal planner.ready_to_select_date_range?, true
+    end
+
+    it "ready_to_generate_dates?" do
+      assert_equal planner.ready_to_generate_dates?, false
+
+      planner.selected_days = [1, 3]
+      assert_equal planner.ready_to_generate_dates?, false
+
+      planner.week_condition = "monthly"
+      planner.monthly_weeks = [2, 4]
+      assert_equal planner.ready_to_generate_dates?, false
+
+      planner.date_range_condition = "episodes"
+      planner.number_of_episodes = 10
+      assert_equal planner.ready_to_generate_dates?, true
+
+      planner.selected_days = []
+      assert_equal planner.ready_to_generate_dates?, false
+
+      planner.selected_days = [1, 3]
+      planner.week_condition = "periodic"
+      assert_equal planner.ready_to_generate_dates?, false
+      planner.period = 1
+      assert_equal planner.ready_to_generate_dates?, true
+
+      planner.date_range_condition = "date"
+      assert_equal planner.ready_to_generate_dates?, false
+      planner.start_date = DateTime.new(2001, 2, 3)
+      planner.end_date = DateTime.new(2001, 2, 4)
+      assert_equal planner.ready_to_generate_dates?, true
+    end
+
+    it "ready_to_generate_drafts?" do
+      assert_equal planner.ready_to_generate_drafts?, false
+
+      dates = []
+      5.times do |i|
+        dates.push(DateTime.new(2001, 2, 3) + i.weeks)
+      end
+      planner.dates = dates
+      assert_equal planner.ready_to_generate_drafts?, false
+
+      planner.publish_time = DateTime.new(2001, 2, 3, 4, 30)
+      assert_equal planner.ready_to_generate_drafts?, false
+
+      planner.segment_count = 2
+      assert_equal planner.ready_to_generate_drafts?, true
+
+      planner.dates = []
+      assert_equal planner.ready_to_generate_drafts?, false
+
+      planner.dates = dates
+      planner.publish_time = nil
+      assert_equal planner.ready_to_generate_drafts?, false
+    end
+  end
+
+  describe "generators" do
```

Comment:
for each test in this suite, i opted to add comments of the dates to assist in readability, since the numbers alone don't necessarily make it apparent.